MigrationInvocationsSafetyTest.migrationCommit_shouldBeRetried_whenTargetNotResponds test failure

https://hazelcast-l337.ci.cloudbees.com/job/Hazelcast-pr-builder/16740/testReport/junit/com.hazelcast.internal.partition/MigrationInvocationsSafetyTest/migrationCommit_shouldBeRetried_whenTargetNotResponds/

**Error Message**
```
expected:<27> but was:<28>
```
**Stacktrace**
```
java.lang.AssertionError: expected:<27> but was:<28>
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.failNotEquals(Assert.java:834)
	at org.junit.Assert.assertEquals(Assert.java:645)
	at org.junit.Assert.assertEquals(Assert.java:631)
	at com.hazelcast.internal.partition.MigrationInvocationsSafetyTest.assertNoDuplicateMigrations(MigrationInvocationsSafetyTest.java:454)
	at com.hazelcast.internal.partition.MigrationInvocationsSafetyTest.migrationCommit_shouldBeRetried_whenTargetNotResponds(MigrationInvocationsSafetyTest.java:268)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:106)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:98)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.lang.Thread.run(Thread.java:748)
```

**Hiccups measured while running test** 
```
'migrationCommit_shouldBeRetried_whenTargetNotResponds(com.hazelcast.internal.partition.MigrationInvocationsSafetyTest):'
15:34:35, accumulated pauses: 316 ms, max pause: 51 ms, pauses over 1000 ms: 0
15:34:40, accumulated pauses: 112 ms, max pause: 5 ms, pauses over 1000 ms: 0
15:34:45, accumulated pauses: 90 ms, max pause: 6 ms, pauses over 1000 ms: 0
15:34:50, accumulated pauses: 85 ms, max pause: 66 ms, pauses over 1000 ms: 0
```

--- a/hazelcast/src/test/java/com/hazelcast/internal/partition/MigrationInvocationsSafetyTest.java
+++ b/hazelcast/src/test/java/com/hazelcast/internal/partition/MigrationInvocationsSafetyTest.java
@@ -276,6 +276,11 @@ public class MigrationInvocationsSafetyTest extends PartitionCorrectnessTestSupp
             final AtomicReference<MigrationInfo> committedMigrationInfoRef = new AtomicReference<MigrationInfo>();
 
             @Override
+            public void onMigrationStart(MigrationParticipant participant, MigrationInfo migrationInfo) {
+               assertClusterStateEventually(ClusterState.ACTIVE, destination);
+            }
+
+            @Override
             public void onMigrationCommit(MigrationParticipant participant, MigrationInfo migrationInfo) {
                 if (participant == MigrationParticipant.DESTINATION && committedMigrationInfoRef.compareAndSet(null, migrationInfo)) {
                     dropOperationsBetween(destination, master, SpiDataSerializerHook.F_ID, singletonList(NORMAL_RESPONSE));

